libraries-3/src/main/java/com/baeldung/mail/attachment/EmailAttachmentReceiver.java

```diff
@@ -0,0 +1,108 @@
+package com.baeldung.mail.attachment;
+
+import static java.util.Arrays.asList;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.NoSuchProviderException;
+import javax.mail.Part;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.internet.MimeBodyPart;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EmailAttachmentReceiver {
+
+    final Logger LOGGER = LoggerFactory.getLogger(EmailAttachmentReceiver.class);
+
+    private static final String MAIL_PROTOCOL_PORT = "mail.%s.port";
+    private static final String MAIL_PROTOCOL_HOST = "mail.%s.host";
+    private static final String MAIL_PROTOCOL_SOCKET_FACTORY_PORT = "mail.%s.socketFactory.port";
+    private static final String MAIL_PROTOCOL_SOCKET_FACTORY_FALLBACK = "mail.%s.socketFactory.fallback";
+    private static final String MAIL_PROTOCOL_SOCKET_FACTORY_CLASS = "mail.%s.socketFactory.class";
+    private static final String FALSE = "false";
+    private static final String JAVAX_NET_SSL_SSL_SOCKET_FACTORY = "javax.net.ssl.SSLSocketFactory";
+    private static final String MULTIPART = "multipart";
+    private static final String INBOX = "INBOX";
+
+    private String protocol;
+    private String host;
+    private String port;
+    private String userName;
+    private String password;
+
+    public EmailAttachmentReceiver(String host, String protocol, String port, String userName, String password) {
+        this.protocol = protocol;
+        this.host = host;
+        this.port = port;
+        this.userName = userName;
+        this.password = password;
+    }
+
+    public void downloadEmailAttachmentsTo(String directory) {
+        Properties properties = new Properties();
+
+        // server settings
+        properties.put(String.format(MAIL_PROTOCOL_HOST, this.protocol), this.host);
+        properties.put(String.format(MAIL_PROTOCOL_PORT, this.protocol), this.port);
+
+        // SSL settings
+        properties.put(String.format(MAIL_PROTOCOL_SOCKET_FACTORY_CLASS, this.protocol), JAVAX_NET_SSL_SSL_SOCKET_FACTORY);
+        properties.put(String.format(MAIL_PROTOCOL_SOCKET_FACTORY_FALLBACK, this.protocol), FALSE);
+        properties.put(String.format(MAIL_PROTOCOL_SOCKET_FACTORY_PORT, this.protocol), this.port);
```

Comment:
As this method is an instance method and not the constructor, the `this` is optional and should probably be removed to reduce the amount for the reader to read.